src/main/java/org/apache/sysml/hops/AggBinaryOp.java

```diff
@@ -344,11 +344,30 @@ protected double computeOutputMemEstimate( long dim1, long dim2, long nnz )
 	protected double computeIntermediateMemEstimate( long dim1, long dim2, long nnz )
 	{
 		double ret = 0;
-		
+
+		if (DMLScript.USE_ACCELERATOR) {
+			// In GPU Mode, these are the intermediate estimates for matmult:
+			// Both dense or both sparse - no intermediate
+			// One sparse, one dense - (dim1 * dim2) OR (dim1 * dim2 + input in sparse format)
+			// In the worst case, an intermediate block other other than a potential temporary sparse-dense block
+			// can be created.
```

Comment:
This comment is confusing as well - what is the difference between an intermediate and temporary block?

There is also a typo ("other other")